File: dotnet/Multiplayer/Rooms/Updates/PollUpdatesResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Rooms.Updates
{
    /// <summary>
    /// Response containing player updates that were targeted to the polling player.
    /// </summary>
    /// <typeparam name="T">The type to deserialize update data into.</typeparam>
    public class PollUpdatesResponse<T> : ApiResponse<ERoomUpdatesPollError> where T : class, new()
    {
        /// <summary>
        /// List of player updates received.
        /// </summary>
        public List<PlayerUpdate<T>> Updates { get; set; } = new();

        /// <summary>
        /// Timestamp of the last update.
        /// </summary>
        public string Last_update { get; set; } = string.Empty;
    }
}

File: java/src/main/java/com/michitai/multiplayer/rooms/updates/PollUpdatesResponse.java
package com.michitai.multiplayer.rooms.updates;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

import java.util.List;

/**
 * Response containing player updates that were targeted to the polling player.
 *
 * @param <T> The type to deserialize update data into.
 */
public class PollUpdatesResponse<T> extends ApiResponse {
    @JsonProperty("updates")
    private List<PlayerUpdate<T>> updates;

    @JsonProperty("last_update")
    private String lastUpdate;

    public List<PlayerUpdate<T>> getUpdates() {
        return updates;
    }

    public void setUpdates(List<PlayerUpdate<T>> updates) {
        this.updates = updates;
    }

    public String getLastUpdate() {
        return lastUpdate;
    }

    public void setLastUpdate(String lastUpdate) {
        this.lastUpdate = lastUpdate;
    }
}
